examples/wallet/test/actions/account.spec.js

```diff
@@ -0,0 +1,168 @@
+        broadcastTransactionCorrectly.firstCall.calledAfter(
+          buildTransactionCorrectly.firstCall
+        );
+      });
+
+      test('AND an action is dispatched to add the transaction to the state', async () => {
+        dispatch.firstCall.calledWith({
+          type: SEND_TRANSACTION,
+          newCounter: counter + 1,
+          id,
+          destination,
+          amount,
+          fee
+        });
+      });
+    });
+
+    describe('WHEN dispatching a sendTransaction action with a mocked, failing transaction builder', () => {
+      const throwWhenBuildingTransaction = stub().returns(
+        Promise.reject(new Error('mock'))
+      );
+      let promise;
+      beforeAll(resetMockHistory);
+      beforeAll(async () => {
+        promise = sendTransaction(
+          throwWhenBuildingTransaction,
+          broadcastTransactionCorrectly
+        )(destination, amount)(dispatch, getState);
+      });
+
+      test('THEN should throw', async () => {
+        return expect(promise).rejects;
+      });
+
+      test('AND the transaction is attempted to be built with the desired parameters', async () => {
+        throwWhenBuildingTransaction.firstCall.calledWith(
+          destination,
+          amount,
+          privateKey,
+          counter,
+          {}
+        );
+      });
+
+      test('AND the transaction is NOT broadcasted', async () => {
+        expect(broadcastTransactionCorrectly.callCount).toBe(0);
+      });
+
+      // TODO: we should notify the user somehow
+      test('AND no action is dispatched', () => {
+        expect(dispatch.callCount).toBe(0);
+      });
+    });
+
+    describe('WHEN dispatching a sendTransaction action with a mocked, failing transaction broadcaster', () => {
+      const throwWhenBroadcastingTransaction = stub().returns(
+        Promise.reject(new Error('mock'))
+      );
+
+      let promise;
+      beforeAll(resetMockHistory);
+      beforeAll(async () => {
+        promise = sendTransaction(
+          buildTransactionCorrectly,
+          throwWhenBroadcastingTransaction
+        )(destination, amount)(dispatch, getState);
+      });
+
+      test('THEN should throw', async () => {
+        return expect(promise).rejects;
+      });
+
+      test('AND the transaction is built with the desired parameters', async () => {
+        expect(
+          buildTransactionCorrectly.firstCall.calledWith(
+            destination,
+            amount,
+            privateKey,
+            counter,
+            {}
+          )
+        ).toBeTruthy();
+      });
+
+      test('AND the transaction is broadcasted', async () => {
+        expect(
+          throwWhenBroadcastingTransaction.firstCall.calledWith(transaction)
+        ).toBeTruthy();
+        expect(
+          throwWhenBroadcastingTransaction.firstCall.calledAfter(
+            buildTransactionCorrectly.firstCall
+          )
+        ).toBeTruthy();
+      });
+
+      // TODO: we should notify the user somehow
```

Comment:
And so, an action *should* actually be dispatched?

Reply:
an action _should_ be dispatched, but to display a notification of some sort, which doesn't apply so far because the app doesn't have notifications